app/src/test/java/com/vero/tasky/auth/presentation/login/LoginViewModelTest.kt

```diff
@@ -0,0 +1,61 @@
+@file:OptIn(ExperimentalCoroutinesApi::class)
+
+package com.vero.tasky.auth.presentation.login
+
+import androidx.lifecycle.SavedStateHandle
+import app.cash.turbine.test
+import com.google.common.truth.Truth.assertThat
+import com.vero.tasky.auth.data.MainCoroutineRule
+import com.vero.tasky.auth.data.repository.AuthRepositoryFake
+import com.vero.tasky.auth.domain.usecase.LoginUseCase
+import com.vero.tasky.auth.domain.usecase.LoginUseCases
+import com.vero.tasky.auth.domain.usecase.ValidateEmailUseCase
+import com.vero.tasky.auth.domain.usecase.password.PasswordValidationResult
+import io.mockk.every
+import io.mockk.mockk
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runTest
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+class LoginViewModelTest {
+
+    @get:Rule
+    val coroutineRule = MainCoroutineRule()
+
+    private lateinit var viewModel: LoginViewModel
+
+    @Before
+    fun setUp() {
+        viewModel = LoginViewModel(
+            savedStateHandle = SavedStateHandle(
+                initialState = mapOf(
+                    "state" to LoginState(
+                        isLoading = false,
+                        isPasswordVisible = false ,
+                        isEmailValid = true,
+                        password = "123",
+                        emailAddress = "123"
+                    )
+                )
+            ),
+            userPreferences = mockk(relaxed = true),
+            loginUseCases = LoginUseCases(
+                validatePasswordUseCase = mockk(relaxed = true) {
+                    every { invoke(password = any()) } returns PasswordValidationResult.SUCCESS },
+                validateEmailUseCase = ValidateEmailUseCase(emailMatcher = mockk(relaxed = true)),
+                loginUseCase = LoginUseCase(AuthRepositoryFake())
+            )
+        )
+    }
+
+    @Test
+    fun `Login, correct info, return results`() = runTest {
```

Comment:
What happens if the email is invalid

Or the password is invalid

Or both

Or none, but the repository returned failure?

You need to test all those cases
Best way imo , is to have an assertion on the viewmodel´s state. Since at the end of the day, that´s what you´ll be showing on the screen